src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import org.slf4j.Logger;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+@SuppressWarnings("java:S1135") // ignore TODOs. will be removed when blob processing part is implemented
+public class ContainerProcessor {
+
+    private static final Logger LOGGER = getLogger(ContainerProcessor.class);
+
+    public void process(BlobContainerAsyncClient containerClient) {
+        String containerName = containerClient.getBlobContainerName();
+        AtomicInteger processedBlobCount = new AtomicInteger(0);
+
+        LOGGER.info("Processing container {}", containerName);
+
+        containerClient
+            .listBlobs()
+            .subscribe(
+                blob -> {
+                    processBlob(blob);
+                    processedBlobCount.incrementAndGet();
+                },
+                null, // TODO: error consumer
+                () -> LOGGER.info(
+                    "Finished processing container {}. Blobs processed: {}", containerName, processedBlobCount.get()
+                )
+            );
+    }
+
+    @SuppressWarnings("java:S1172")
+    // ignore unused variable. This will be removed once the blob processor code is implemented
+    private void processBlob(BlobItem blob) {
+        // TODO: process blob
```

Comment:
This change should be done before this PR, i.e. dependencies should be ready first, rather than having some orchestration that doesn't do anything. Otherwise, we have a feature, but we don't have proper tests for it, because it doesn't work yet. Then we add all necessary dependencies, write all tests for them and consider the story done, without realising that there are tests missing at higher level.